src/Core/Rexl.Bind/Operations/Funcs/Text.cs

```diff
@@ -758,3 +758,123 @@ public static string Exec(string src, string remove, string insert)
         return src.Replace(remove, insert);
     }
 }
+
+/// <summary>
+/// Functions to add padding to either end of a string.
+/// The left version (PadLeft) will add spaces to the left of the given string.
+/// The right version (PadRight) will add spaces to the right of the given string.
+/// </summary>
+public sealed partial class TextPadFunc : RexlOper
+{
+    public enum PadKind : byte
+    {
+        // Trim both leading and trailing whitespace.
+        Center,
+        // Trim only leading whitespace.
+        Start,
+        // Trim only trailing whitespace.
+        End
+    }
+
+    public static readonly TextPadFunc PadCenter = new TextPadFunc(PadKind.Center, "PadCenter");
+    public static readonly TextPadFunc PadStart = new TextPadFunc(PadKind.Start, "PadStart");
+    public static readonly TextPadFunc PadEnd = new TextPadFunc(PadKind.End, "PadEnd");
+
+    public PadKind Kind { get; }
+
+    public Func<string, long, string> Map { get; }
+
+    private TextPadFunc(PadKind kind, string name)
+        : base(isFunc: true, new DName(name), BindUtil.TextNs, 2, 2)
+    {
+        switch(kind) {
```

Comment:
In Visual Studio (aka VS), use autoformat (Ctrl+K+D with the settings I use) to fix the braces. The open brace goes on its own line.